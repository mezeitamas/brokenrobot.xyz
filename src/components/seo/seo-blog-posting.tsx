import React from 'react';
import type { FunctionComponent, PropsWithChildren } from 'react';

import { useSiteMetadata } from '../../hooks/use-site-metadata';

import { MetaOgArticle } from './meta-og-article/meta-og-article';
import { MetaTwitter } from './meta-twitter/meta-twitter';
import { RichResultsBlogPosting } from './rich-results-blog-posting/rich-results-blog-posting';

type SeoBlogPostingProps = {
    title: string;
    description: string;
    pathname: string;
    published: string;
    tags: string[];
};

const SeoBlogPosting: FunctionComponent<PropsWithChildren<SeoBlogPostingProps>> = ({
    title,
    description,
    pathname,
    published,
    tags,
    children
}) => {
    const { title: siteName, siteUrl, author } = useSiteMetadata();
    const url = `${siteUrl}${pathname}`;

    return (
        <>
            <html lang="en" />

            <title>{title}</title>

            <meta
                httpEquiv="cache-control"
                content="public, max-age=0, must-revalidate"
            />

            <meta
                name="description"
                content={description}
            />

            <meta
                name="keywords"
                content={tags.join(', ')}
            />

            <MetaOgArticle
                siteName={siteName}
                title={title}
                description={description}
                url={url}
                published={published}
                tags={tags}
                author={author.name}
            />

            <RichResultsBlogPosting
                siteName={siteName}
                siteUrl={siteUrl}
                title={title}
                description={description}
                url={url}
                published={published}
                author={author}
            />

            <MetaTwitter
                title={title}
                description={description}
                url={url}
            />

            {children}
        </>
    );
};

export { SeoBlogPosting };

export type { SeoBlogPostingProps };
