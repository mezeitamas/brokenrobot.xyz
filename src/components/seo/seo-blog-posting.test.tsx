import React from 'react';

import { create } from 'react-test-renderer';

import { SeoBlogPosting } from './seo-blog-posting';
import type { SeoBlogPostingProps } from './seo-blog-posting';

jest.mock('../../hooks/use-site-metadata', () => {
    return {
        useSiteMetadata: jest.fn().mockImplementation(() => {
            return {
                title: 'Broken Robot',
                description:
                    'Personal website and blog of Tamas Mezei. Welcome to my little corner of the web, where I share my professional experiences, thoughts, adventures, and projects with the world.',
                siteUrl: 'https://www.brokenrobot.xyz',
                author: {
                    name: 'Tamas Mezei',
                    social: {
                        github: 'https://github.com/mezeitamas',
                        linkedin: 'https://www.linkedin.com/in/mezeitamas/'
                    }
                }
            };
        })
    };
});

describe('<SeoBlogPosting />', () => {
    test('should render', () => {
        // Given
        const props: SeoBlogPostingProps = {
            title: 'Hosting a static website on Amazon S3',
            description:
                'Hosting a simple static website on Amazon Web Services (AWS) can be daunting at first, but luckily itâ€™s quite straightforward. One option...',
            pathname: '/blog/hosting-a-static-website-on-amazon-s3/',
            published: '2023-05-04T17:58:32.000Z',
            tags: ['aws', 's3', 'route53', 'staticwebsite', 'hosting']
        };

        // When
        const component = create(<SeoBlogPosting {...props} />).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });
});
