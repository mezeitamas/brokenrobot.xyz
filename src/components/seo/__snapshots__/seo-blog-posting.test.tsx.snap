// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SeoBlogPosting /> should render 1`] = `
[
  <html
    lang="en"
  />,
  <title>
    Hosting a static website on Amazon S3
  </title>,
  <meta
    content="public, max-age=0, must-revalidate"
    httpEquiv="cache-control"
  />,
  <meta
    content="Hosting a simple static website on Amazon Web Services (AWS) can be daunting at first, but luckily it’s quite straightforward. One option..."
    name="description"
  />,
  <meta
    content="aws, s3, route53, staticwebsite, hosting"
    name="keywords"
  />,
  <meta
    content="article"
    property="og:type"
  />,
  <meta
    content="en_US"
    property="og:locale"
  />,
  <meta
    content="Broken Robot"
    property="og:site_name"
  />,
  <meta
    content="Hosting a static website on Amazon S3"
    property="og:title"
  />,
  <meta
    content="Hosting a simple static website on Amazon Web Services (AWS) can be daunting at first, but luckily it’s quite straightforward. One option..."
    property="og:description"
  />,
  <meta
    content="https://www.brokenrobot.xyz/blog/hosting-a-static-website-on-amazon-s3/"
    property="og:url"
  />,
  <meta
    content="2023-05-04T17:58:32.000Z"
    property="og:article:published_time"
  />,
  <meta
    content="aws"
    property="og:article:tag"
  />,
  <meta
    content="s3"
    property="og:article:tag"
  />,
  <meta
    content="route53"
    property="og:article:tag"
  />,
  <meta
    content="staticwebsite"
    property="og:article:tag"
  />,
  <meta
    content="hosting"
    property="og:article:tag"
  />,
  <meta
    content="Tamas Mezei"
    property="og:article:author"
  />,
  <script
    type="application/ld+json"
  >
    {"@context":"https://schema.org","@type":"BlogPosting","@id":"https://www.brokenrobot.xyz/blog/hosting-a-static-website-on-amazon-s3/#BlogPosting","headline":"Hosting a static website on Amazon S3","name":"Hosting a static website on Amazon S3","description":"Hosting a simple static website on Amazon Web Services (AWS) can be daunting at first, but luckily it’s quite straightforward. One option...","datePublished":"2023-05-04T17:58:32.000Z","dateModified":"2023-05-04T17:58:32.000Z","url":"https://www.brokenrobot.xyz/blog/hosting-a-static-website-on-amazon-s3/","inLanguage":"en-US","author":[{"@type":"Person","@id":"https://www.brokenrobot.xyz/about-me/#Person","name":"Tamas Mezei","url":"https://www.brokenrobot.xyz/about-me/","sameAs":["https://github.com/mezeitamas","https://www.linkedin.com/in/mezeitamas/"]}],"publisher":{"@type":"Organization","@id":"https://www.brokenrobot.xyz/#Organization","name":"Broken Robot","url":"https://www.brokenrobot.xyz"},"isPartOf":{"@type":"WebSite","@id":"https://www.brokenrobot.xyz/#WebSite","name":"Broken Robot","url":"https://www.brokenrobot.xyz","inLanguage":"en-US","author":[{"@type":"Person","@id":"https://www.brokenrobot.xyz/about-me/#Person","name":"Tamas Mezei","url":"https://www.brokenrobot.xyz/about-me/","sameAs":["https://github.com/mezeitamas","https://www.linkedin.com/in/mezeitamas/"]}],"publisher":{"@type":"Organization","@id":"https://www.brokenrobot.xyz/#Organization","name":"Broken Robot","url":"https://www.brokenrobot.xyz"}}}
  </script>,
  <meta
    content="summary"
    name="twitter:card"
  />,
  <meta
    content="Hosting a static website on Amazon S3"
    name="twitter:title"
  />,
  <meta
    content="Hosting a simple static website on Amazon Web Services (AWS) can be daunting at first, but luckily it’s quite straightforward. One option..."
    name="twitter:description"
  />,
  <meta
    content="https://www.brokenrobot.xyz/blog/hosting-a-static-website-on-amazon-s3/"
    name="twitter:url"
  />,
]
`;
